src/app/app-components/courses/add-course/add-course.component.ts

```diff
@@ -72,16 +92,13 @@ export class AddCourseComponent implements OnInit {
     this.router.navigate(['/courses']);
   }
 
-  onDateChange(date: Date) {
-    this.date = date;
-  }
-
-  onDurationChange(duration: number) {
-    this.length = duration;
+  private parseDateString(dateString: string): string {
+    const date = new Date(dateString);
```

Comment:
Be careful with the time zone when instantiating a Date object passing a string in the format 'mm/dd/yyyy'. It defaults to UTC time zone instead of local time.

Reply:
Good catch! - I'll fix this before submitting.